### PR TITLE
Add support for mounting custom query modules

### LINKED ISSUE #66: Injecting custom query modules
Hi, as the title suggests, I'd like to add the capability to mount custom query modules into the app during the deployment. I know one way would be to bake these files into the Docker image but I'd prefer to keep them separate to avoid having to maintain one extra component.

Would you be interested in a small contribution? I have the code almost ready :)

### PR DESCRIPTION
**Pull request by matkob**
Closes #66

### CODE REVIEW

File: charts/memgraph/values.yaml [hunk 1]
```diff
@@ -135,3 +135,12 @@ container:
...
       failureThreshold: 30
+
+# List of custom query modules to be mounted into the app.
+# These will be loaded automatically, on startup.
+# Each module must be exposed by a ConfigMap under a specific file name.
+customQueryModules: []
+#   Must be an existing ConfigMap
+# - volume: ""
+#   Must be present in the ConfigMap referenced with `volume`
+#   file: ""
```
[the remark this answers is not in the document]
Comment (antejavor) [reply charts/memgraph/values.yaml:146]: We need to fix this so the action can be re-run.

**Review by antejavor — CHANGES_REQUESTED** [review]
Hi @matkob thanks a lot  for this contribution. 🚀  Let's just fix the empty space, so we can re-run the test.

If I understood correctly, you would have something like this: 

values.yaml
``` 
customQueryModules:
  - volume: "module-volume-1"
    file: "module1.py"
  - volume: "module-volume-2"
    file: "module2.py"
```
Template:
```
- name: "module-volume-1"
  mountPath: "/var/lib/memgraph/internal_modules/module1.py"
  subPath: "module1.py"
- name: "module-volume-2"
  mountPath: "/var/lib/memgraph/internal_modules/module2.py"
  subPath: "module2.py"
```

We obviously do not have a clear way of adding this,  except maybe [user defined volume here](https://github.com/memgraph/helm-charts/blob/main/charts/memgraph/templates/statefulset.yaml#L185-L188). 

I am curious if you can comment on what is the benefit of having a volume per QM for you? 

Just asking, since I didn't have a chance to use QMs + k8s, so I am curious about the flow and why did you configure it this way.

File: charts/memgraph/values.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (matkob) [reply charts/memgraph/values.yaml]: Ahh, sorry about that. Added the line

**Comment by matkob** [comment]
Hi @antejavor, good question! Initially I wanted to create and mount only one ConfigMap with all the custom query modules inside, potentially even stored in multiple files. Unfortunately I bumped into issues related to how Kubernetes mounts the volumes - [it adds timestamp to the mountPath and creates a symlink](https://stackoverflow.com/questions/68331724/kubernetes-volumemount-path-contains-timestamp). This would end up in QM not working in memgraph (they would be duplicated and memgraph wouldn't load the code properly).

Mounting each QM separately, using subPath, fixed the issue and the workaround didn't seem "too dirty" to me. Let me know if this seems sound to you, too :)

**Review by antejavor — APPROVED** [review]
@matkob cool 🚀  thanks a lot for this. 

Let's keep this as a default approach for now, I will take a look at the other options at the future.

**Pull request merged**